src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -229,5 +229,35 @@ internal static void ValidateNumber(ReadOnlySpan<byte> utf8FormattedNumber)
                     nameof(utf8FormattedNumber));
             }
         }
+
+        public static readonly UTF8Encoding s_utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
```

Comment:
Catching UTF8Encoding like this is slight de-optimization. ` Encoding.UTF8` has same cache, and it also returns internal more specialized sealed `UTF8Encoding` type that allows de-virtualization to kick-in in more situations.